Common++/src/IpAddress.cpp

```diff
@@ -43,6 +48,46 @@ namespace pcpp
 	}
 
 
+	bool IPv4Address::matchSubnet(const std::string& subnet) const
+	{
+		std::stringstream ss(subnet);
+		std::string subnetOnly, subnetPrefixStr;
+		std::getline(ss, subnetOnly, '/');
+		std::getline(ss, subnetPrefixStr);
+
+		if (subnetPrefixStr.empty() || !std::all_of(subnetPrefixStr.begin(), subnetPrefixStr.end(), ::isdigit)) {
+			PCPP_LOG_ERROR("subnet prefix '" << subnetPrefixStr << "' must be an integer");
+			return false;
+		}
+
+		uint32_t subnetPrefix;
+		try {
+			subnetPrefix = std::stoi(subnetPrefixStr);
```

Comment:
I noticed that if `subnetPrefixStr` is too large `std:stoi` throws `std::out_of_range` exception, here is an example:
https://www.online-cpp.com/JuthUqrVyI

We probably want to catch that too...